src/lib/apy/derivativeScrt.ts

```diff
@@ -0,0 +1,175 @@
+import {
+  SecretChainDataQueryModel,
+  SecretQueryOptions,
+  ValidatorRate,
+} from '~/types/apy';
+import { forkJoin, lastValueFrom, map } from 'rxjs';
+import {
+  DerivativeScrtValidator,
+  DerivativeScrtInfo,
+} from '~/types/contracts/derivativeScrt/model';
+import { convertCoinFromUDenom } from '~/lib/utils';
+import { queryDerivativeScrtInfo$ } from '~/contracts/services/derivativeScrt';
+import { secretChainQueries$ } from './secretQueries';
+
+const SECRET_DECIMALS = 6;
+
+/**
+* Get single validator commission rate from a list of all validators
+* @example Example usage of getValidatorCommission
+* // returns 0.05 if commission was 5%
+*/
+function getValidatorCommission(
+  validatorAddress: string,
+  validatorList: ValidatorRate[],
+):number {
+  const result = validatorList.filter((
+    validator,
+  ) => validator.validatorAddress === validatorAddress);
+  if (result[0]) {
+    const commission = Number(result[0].ratePercent);
+    return commission;
+  }
+  throw new Error(`Error: validator address ${validatorAddress} not found in list`);
+}
+
+/**
+* Calculate an Aggregate Staking Rewards Return Rate based on commission rates and weights assigned
+* to each validator and their commission rates.
+*/
+function calcAggregateAPR({
+  networkValidatorList,
+  validatorSet,
+  inflationRate,
+  totalScrtStaked,
+  totalScrtSupply,
+  foundationTax,
+  communityTax,
+}:
+{
+  networkValidatorList:ValidatorRate[],
+  validatorSet: DerivativeScrtValidator[],
+  inflationRate:number,
+  totalScrtStaked:number,
+  totalScrtSupply:number,
+  foundationTax:number,
+  communityTax:number,
+}) {
+  let aggregateApr = 0;
+  // typically total weight will equal 100, but in rare situations the stkd-SCRT contract
+  // will automatically remove validators from the active set and the total will not add up to 100
+  const totalWeight = validatorSet.reduce((acc, validator) => acc + validator.weight, 0);
+
+  validatorSet.forEach((validator) => {
+    // Get commission rate for a single validator
+    const commissionRate = getValidatorCommission(validator.validatorAddress, networkValidatorList);
+    // Calculate APR of a single validator
+    const apr = (inflationRate / (totalScrtStaked / totalScrtSupply))
```

Comment:
I know this is copy/paste from front end, but seems like it will be better to move this into it's own function with a unit test on it.